src/DropdownReferenceSet/components/DropdownReferenceSetContainer.ts

```diff
@@ -0,0 +1,258 @@
+            searchText: this.props.searchText,
+            loadingText: this.props.loadingText,
+            minimumCharacter: this.props.minimumCharacter,
+            showLabel: this.props.showLabel,
+            styleObject: parseStyle(this.props.style)
+        });
+    }
+
+    componentWillReceiveProps(newProps: ContainerProps) {
+        if (newProps.mxObject && (newProps.mxObject !== this.props.mxObject)) {
+            const selected = newProps.mxObject.get(this.association) as string;
+            this.getSelectedValues(newProps);
+            this.resetSubscriptions(newProps.mxObject);
+            this.retrieveOptions(newProps);
+            this.setState({ selected });
+        } else {
+            this.setState({ selected: [] });
+        }
+    }
+
+    componentDidMount() {
+        initializeReactFastclick();
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+    }
+
+    private isReadOnly = (): boolean => {
+        const { editable, mxObject, readOnly, attribute } = this.props;
+
+        return editable !== "default" || (!mxObject || readOnly || !(attribute && mxObject.isReadonlyAttr(attribute)));
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: () => this.getSelectedValues(this.props),
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.association,
+                callback: () => this.getSelectedValues(this.props),
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.attribute,
+                callback: () => this.getSelectedValues(this.props),
+                guid: mxObject.get(this.association) as string
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                entity: this.props.entityPath.split("/")[1],
+                callback: () => {
+                    this.getSelectedValues(this.props);
+                    this.retrieveOptions(this.props);
+                }
+            }));
+        }
+    }
+
+    private onChange(recentSelection: ReferenceOption[] | any) {
+        if (!this.props.mxObject) {
+            return;
+        }
+
+        const selectedOptions: string[] = [];
+        const previousSelection: string[] = [];
+
+        recentSelection.forEach((selection: ReferenceOption) => {
+            selectedOptions.push(selection.value as string);
+        });
+
+        if (this.state.selected.length > 0) {
+            this.state.selected.forEach((selection: ReferenceOption) => {
```

Comment:
Use map instead of foreach with push inside. And remove the previousSelection object reference.

`const map = this.state.selected.map((selection: ReferenceOption) => selection.value as string);
this.props.mxObject.removeReferences(this.association, map);`

Reply:
Done